lib/matplotlib/tests/test_patches.py

```diff
@@ -149,6 +149,40 @@ def test_rotate_rect_draw(fig_test, fig_ref):
     assert rect_test.get_angle() == angle
 
 
+@check_figures_equal(extensions=['png'])
+def test_dash_offset_patch_draw(fig_test, fig_ref):
+    ax_test = fig_test.add_subplot()
+    ax_ref = fig_ref.add_subplot()
+
+    loc = (0, 0)
+    width, height = (1, 1)
+    edgecolor = 'b'
+    linestyle = (0, [6, 6])
+    linestyle_hacked = (0, [0, 6, 6, 0])
+    rect_ref = Rectangle(loc, width, height, edgecolor=edgecolor,
+                                                linestyle=linestyle)
+    rect_ref2 = Rectangle(loc, width, height, edgecolor=edgecolor,
+                                        linestyle=linestyle_hacked)
+    ax_ref.add_patch(rect_ref)
+    ax_ref.add_patch(rect_ref2)
+
+    assert rect_ref.get_linestyle() == linestyle
+    assert rect_ref2.get_linestyle() == linestyle_hacked
```

Comment:
```suggestion
    assert rect_ref.get_linestyle() == linestyle
    assert rect_ref2.get_linestyle() == linestyle_hacked
    ax_ref.add_patch(rect_ref)
    ax_ref.add_patch(rect_ref2)
```
Use the same order of the statements in ref and test sections. You could alternatively change the order in the ref section. I have a very slight preference of doing the asserts first. We communicate with that that we are only interested in the property of the created patches and that `add_patch` is not relevant for that. - Readers might otherwise ponder whether the `add_patch` plays some roll if we delay the assert behind that.